lib/beaker/test_suite_result.rb

```diff
@@ -0,0 +1,246 @@
+            [
+              ['name' , @name],
+              ['tests', test_count],
+              ['errors', errored_tests],
+              ['failures', failed_tests],
+              ['skipped', skipped_tests],
+              ['pending', pending_tests],
+              ['total', @total_tests],
+              ['time', "%f" % (stop_time - start_time)]
+          ])
+          properties = suite.add_element(REXML::Element.new('properties'))
+          @options.each_pair do |name,value|
+            property = properties.add_element(REXML::Element.new('property'))
+            property.add_attributes([['name', name], ['value', value.to_s || '']])
+          end
+
+          test_cases_to_report = @test_cases
+          test_cases_to_report = @test_cases.sort { |x,y| y.runtime <=> x.runtime } if time_sort
+          test_cases_to_report.each do |test|
+            item = suite.add_element(REXML::Element.new('testcase'))
+            item.add_attributes(
+              [
+                ['classname', File.dirname(test.path)],
+                ['name', File.basename(test.path)],
+                ['time', "%f" % test.runtime]
+              ])
+
+            test.exports.each do |export|
+              export.keys.each do |key|
+                item.add_attribute(key.to_s.tr(" ", "_"), export[key])
+              end
+            end
+
+            #Report failures
+            if test.test_status == :fail || test.test_status == :error
+              status = item.add_element(REXML::Element.new('failure'))
+              status.add_attribute('type', test.test_status.to_s)
+              if test.exception
+                status.add_attribute('message', test.exception.to_s.gsub(/\e/,''))
+                data = LoggerJunit.format_cdata(test.exception.backtrace.join('\n'))
+                REXML::CData.new(data, true, status)
+              end
+            end
+
+            if test.test_status == :skip
+              status = item.add_element(REXML::Element.new('skipped'))
+              status.add_attribute('type', test.test_status.to_s)
+            end
+
+            if test.test_status == :pending
+              status = item.add_element(REXML::Element.new('pending'))
+              status.add_attribute('type', test.test_status.to_s)
+            end
+
+            if test.sublog
+              stdout = item.add_element(REXML::Element.new('system-out'))
+              data = LoggerJunit.format_cdata(test.sublog)
+              REXML::CData.new(data, true, stdout)
+            end
+
+            if test.last_result and test.last_result.stderr and not test.last_result.stderr.empty?
+              stderr = item.add_element('system-err')
+              data = LoggerJunit.format_cdata(test.last_result.stderr)
+              REXML::CData.new(data, true, stderr)
+            end
+          end
+        end
+      rescue Exception => e
+        @logger.error "failure in XML output: \n#{e.to_s}" + e.backtrace.join("\n")
+      end
+    end
+
+  end
+end
```

Comment:
i can see why we'd want to move this class for better file layout, etc.  

if it were in a separate commit it would make the diffs easier to read.